# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/ThirdPartyCookieBlockingExceptionListService.sys.mjs
# Commit: de6a26d3e3df
# Full Hash: de6a26d3e3dfd5d09121f11ae2b09903c780179b
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2024-12-17 17:00:49
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 4: Implement the third-party cookie blocking excpetion in CookieService. r=cookie-reviewers,baku,valentin
#   
#   The patch integrates the third-party cookie blocking excpetion in
#   CookieService.
#   
# ==============================================================================

diff -r b2934cae3c23 -r de6a26d3e3df netwerk/cookie/ThirdPartyCookieBlockingExceptionListService.sys.mjs
--- a/netwerk/cookie/ThirdPartyCookieBlockingExceptionListService.sys.mjs	Tue Dec 17 08:57:30 2024 +0000
+++ b/netwerk/cookie/ThirdPartyCookieBlockingExceptionListService.sys.mjs	Tue Dec 17 08:57:31 2024 +0000
@@ -57,10 +57,10 @@
 
   #handleExceptionChange(created = [], deleted = []) {
     if (created.length) {
-      // TODO: Calling CookieService API to remove exception sites.
+      Services.cookies.addThirdPartyCookieBlockingExceptions(created);
     }
     if (deleted.length) {
-      // TODO: Calling CookieService API to add exception sites.
+      Services.cookies.removeThirdPartyCookieBlockingExceptions(deleted);
     }
   }
 